apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/file/EnumAsEnumBuilder.kt

```diff
@@ -49,7 +50,9 @@ class EnumAsEnumBuilder(
         .addProperty(rawValuePropertySpec)
         .addType(companionTypeSpec())
         .apply {
-          values.forEach { value -> addEnumConstant(layout.enumName(value.name), value.enumConstTypeSpec()) }
+          values.forEach { value ->
+            addEnumConstant(layout.enumValueName(value.name).escapeKotlinReservedEnumValueNames(), value.enumConstTypeSpec())
```

Comment:
Nit: I'd like the escaping to happen all in `CodegenLayout`. The layout takes GraphQL in and outputs Kotlin. So it certainly needs both `layout.enumValueName(value.name)` and `layout.sealedClassValueName(value.name)`